service/history/task/transfer_active_task_executor.go

```diff
@@ -1054,7 +1054,7 @@ func (t *transferActiveTaskExecutor) signalExternalExecutionCompleted(
 	err := t.updateWorkflowExecution(ctx, wfContext, true,
 		func(ctx context.Context, mutableState execution.MutableState) error {
 			if !mutableState.IsWorkflowExecutionRunning() {
-				return &types.EntityNotExistsError{Message: "Workflow execution already completed."}
+				return &types.WorkflowExecutionAlreadyCompletedError{Message: "Workflow execution already completed."}
```

Comment:
Do we need to update the code on L1076? Same for other changes.

Reply:
Good catch! I didn't realize we have special treatments for this error in this file.